### PR TITLE
Epic 8: Notification UI Updates [ready to merge]

### PR DESCRIPTION
**Pull request by ackernaut**
This PR adds styles for #1479 (notification UI updates) and #1430 (project notifications).

## Test Plan
To fully test notification UI (9 different notifications), you need to sign in as 2 different users locally:
- [ ] Invite the other local user to a team. As that user, check the presentation and functionality of the team invite notification.
- [ ] As a non-billing leader, invite 2 non-org users to a team. As the billing leader, check the presentation and functionality of the invite request notification.
- [ ] As a billing leader, approve a team invitation of a non-org user. As the user who sent the invitation, check the presentation and functionality of the approval notification.
- [ ] As a billing leader, reject a team invitation of a non-org user. As the user who sent the invitation, check the presentation and functionality of the rejection notification.
- [ ] As a billing leader, promote the other user to billing leader. As the other user, check the presentation and functionality of the billing leader promotion notification.
- [ ] As a user, mention another user in a project. As the other user, check the presentation and functionality of the mention notification.
- [ ] As a user, assign a project to another user. As the other user, check the presentation and functionality of the assignment notification.
- [ ] Archive a team. Check the presentation and functionality of the archived team notification.
- [ ] A faked payment notification was checked locally for presentation and functionality.
- [ ] The “Clear All” control (top right) clears all notifications that do not require an explicit action. The control does not show when there are not any clear-able notifications.

### CODE REVIEW

**Comment by ackernaut** [comment]
ok @mattkrick this is ready for a review:
- I used handlers similar to the buttons to make the team or org names clickable in the notification copy.
- I think we should filter out notifications when a user is mentioned but another owns the project and the private tag is present [see comment here](https://github.com/ParabolInc/action/issues/1363#issuecomment-344636501)
- the “Clear All” button is ready for wiring up, but could be another branch if desired.
- I think we can defer on clickable links in readOnly content for now ([see here](https://github.com/ParabolInc/action/issues/1458#issuecomment-344408623))

**Comment by ackernaut** [comment]
A screenshot with latest styles:
image

File: src/universal/components/SocketRoute/SocketRoute.js [hunk 1]
```diff
@@ -69,6 +69,10 @@ const query = graphql`
               project {
                 id
                 content
+                teamMember {
```
Comment (mattkrick) [inline src/universal/components/SocketRoute/SocketRoute.js]: this won't be functional until i merge my behemoth of a `projects-to-relay` PR

File: src/universal/modules/notifications/components/Notifications/Notifications.js [hunk 2]
```diff
@@ -4,22 +4,49 @@ import React from 'react';
...
 import withStyles from 'universal/styles/withStyles';
+import Button from 'universal/components/Button/Button';
 
 const Notifications = (props) => {
   const {
     dispatch,
     notifications,
     styles
   } = props;
+  const paymentFake = {
```
Comment (mattkrick) [inline src/universal/modules/notifications/components/Notifications/Notifications.js]: lets nuke this for prod

File: src/universal/modules/notifications/components/ProjectInvolves.js [hunk 3]
```diff
@@ -48,48 +50,63 @@ const ProjectInvolves = (props) => {
...
+          </span>
+          <span>{':'}</span>
         </div>
-        <div className={css(styles.projectListView)}>
+        <div className={projectStyles}>
           <Editor
             readOnly
             editorState={editorState}
           />
+          <div className={css(styles.owner)}>
```
Comment (mattkrick) [inline src/universal/modules/notifications/components/ProjectInvolves.js]: so this isn't blocked by the `projects-to-relay` branch, let's wrap this in a `{teamMember && ...}` block so it's OK if the teamMember is null

**Comment by ackernaut** [comment]
## Test Cycle No. 1 (TA)
### Status: Completed!
To fully test notification UI (9 different notifications), you need to sign in as 2 different users locally:
- [x] Invite the other local user to a team. As that user, check the presentation and functionality of the team invite notification.
- [x] As a non-billing leader, invite 2 non-org users to a team. As the billing leader, check the presentation and functionality of the invite request notification.
- [x] As a billing leader, approve a team invitation of a non-org user. As the user who sent the invitation, check the presentation and functionality of the approval notification.
- [x] As a billing leader, reject a team invitation of a non-org user. As the user who sent the invitation, check the presentation and functionality of the rejection notification.
- [x] As a billing leader, promote the other user to billing leader. As the other user, check the presentation and functionality of the billing leader promotion notification.
- [x] As a user, mention another user in a project. As the other user, check the presentation and functionality of the mention notification.
- [x] As a user, assign a project to another user. As the other user, check the presentation and functionality of the assignment notification.
- [x] Archive a team. Check the presentation and functionality of the archived team notification.
- [x] A faked payment notification was checked locally for presentation and functionality.

**Comment by ackernaut** [comment]
@mattkrick I made the changes you suggested. This is ready to be merged as-is or we can look at the items in [this comment above](https://github.com/ParabolInc/action/pull/1489#issuecomment-344655525). ( @dan-f )

File: src/universal/modules/notifications/components/Notifications/Notifications.js [hunk 4]
```diff
@@ -1,21 +1,47 @@
...
+    orgId: 'BkeCjMNHpZ'
+  };
+
+  const clearAllNotifications = () =>
+    Promise.all(
+      notifications.edges
+        .map(({node: notification}) => notification)
+        .filter((n) => !requiresAction(n))
+        .map(({id}) => {
+          console.log('clearing: ', id, fromGlobalId(id))
```
Comment (ackernaut) [inline src/universal/modules/notifications/components/Notifications/Notifications.js]: clean up `console.log` for prod?

File: src/universal/data/notification.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    | PROMOTE_TO_BILLING_LEADER
+    | REQUEST_NEW_USER
+    | TEAM_ARCHIVED
+    | TEAM_INVITE,
+  userIds: string[]
+};
+
+// Note: this is a conservative estimate; not all of these notifications "require" action,
+// but they all offer an action other than to clear the notification.
+const NOTIFICATION_TYPES_REQUIRING_ACTION = new Set([
```
Comment (ackernaut) [inline src/universal/data/notification.js]: I think we should scope this down to: `PAYMENT_REJECTED`, `REQUEST_NEW_USER`, and `TEAM_INVITE` as the others can be clearable and the other action is options (see board, team, org).

Comment (ackernaut) [reply src/universal/data/notification.js]: Updated the list, feel free to push back if you want, but the design intent in my mind was that when there was a clear button with another action button, the other action was optional, not required.

File: package.json [hunk 6]
```diff
@@ -36,11 +36,12 @@
     "db:migrate-down": "node ./src/server/database/migrate.babel.js down",
     "db:migrate-down-all": "node ./src/server/database/migrate.babel.js down --all",
     "debug:stripe": "node ./src/server/billing/debug.babel.js",
+    "flow": "flow",
```
Comment (ackernaut) [inline package.json]: @dan-f seems we need to have a `.flowconfig` for the build to pass? Digging in deeper…

File: src/universal/data/notification.js (hunk 5 above)
Comment (dan-f) [reply src/universal/data/notification.js]: That's perfect! I'm going to update the comment for accuracy.

File: package.json (hunk 6 above)
Comment (dan-f) [reply package.json]: Oh shoot. I thought I added that. Hold on, I've got it.

**Comment by ackernaut** [comment]
## Test Cycle No. 2 (TA)
### Status: Pending
To fully test notification UI (9 different notifications), you need to sign in as 2 different users locally:
- [ ] Invite the other local user to a team. As that user, check the presentation and functionality of the team invite notification.
- [ ] As a non-billing leader, invite 2 non-org users to a team. As the billing leader, check the presentation and functionality of the invite request notification.
- [ ] As a billing leader, approve a team invitation of a non-org user. As the user who sent the invitation, check the presentation and functionality of the approval notification.
- [ ] As a billing leader, reject a team invitation of a non-org user. As the user who sent the invitation, check the presentation and functionality of the rejection notification.
- [ ] As a billing leader, promote the other user to billing leader. As the other user, check the presentation and functionality of the billing leader promotion notification.
- [ ] As a user, mention another user in a project. As the other user, check the presentation and functionality of the mention notification.
- [ ] As a user, assign a project to another user. As the other user, check the presentation and functionality of the assignment notification.
- [ ] Archive a team. Check the presentation and functionality of the archived team notification.
- [ ] A faked payment notification was checked locally for presentation and functionality.
- [ ] The “Clear All” control (top right) clears all notifications that do not require an explicit action.

**Comment by ackernaut** [comment]
ok @mattkrick @dan-f this is passing, tested, and ready for merge!

**Comment by dan-f** [comment]
👍 👏  woohoo! @ackernaut you want to do the honors & merge?

**Pull request merged**